server/routes/posts.js

```diff
@@ -1,32 +1,171 @@
 const express = require('express');
+const auth = require('../middleware/auth');
+const errorCatcher = require('../middleware/errorCatcher');
+const {
+	body,
+	validationResult,
+} = require('express-validator');
+const ObjectId = require('mongoose').Types.ObjectId;
 const router = express.Router();
 
+const User = require('../models/User');
+const Post = require('../models/Post');
+
 // @route   GET api/posts
 // @desc    Get all users posts
 // @access  Private
-router.get('/', (req, res) => {
-	res.send('Get all users posts');
-});
+router.get(
+	'/',
+	auth,
+	errorCatcher(async (req, res) => {
+		const posts = await Post.find({
+			author: req.user.id,
+		}).sort({
+			date: -1,
```

Comment:
What does `-1` mean in this context? Is it descending sort?

Reply:
Yes :). I'll leave it as is, but I wonder, is something like this preferable for the sake of readability?
```suggestion
			date: 'desc',
```

Reply:
Yeah, `'desc'` would help readability a ton.